src/core/Params.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * Handles placeholder replacement with given params.
+ */
+export default class Params {
+    /**
+     * @param {Object} params
+     */
+    constructor(params) {
+        this.params = params;
+        this.index = 0;
+    }
+
+    /**
+     * Returns param value that matches given placeholder with param key.
+     * @param {String} placeholder
+     * @return {String} param
+     */
+    get(placeholder) {
+        if (!this.params) {
+            return placeholder;
+        }
+        const matches = placeholder.match(/\w+(\s+\w+)*/);
```

Comment:
I don't really understand why this regex is the way it is.

I understand that the goal is to detect whether it's a named placeholder. However it doesn't quite work - for example when I have a placeholder that does not consist of just spaces and word characters:

    SELECT @'$^&*'

It might be simpler to differentiate between named and indexed placeholders inside tokenizer.

Another thing to consider is escape sequences inside strings:

    SELECT @"hello \" world"